Extensions/addressbar.js

```diff
@@ -0,0 +1,80 @@
+//* TITLE Addressbar Updater **//
+//* VERSION 1.0.1 **//
+//* DESCRIPTION Keeps the url in its place **//
+//* DEVELOPER srsutherland **//
+//* DETAILS Sets the url to the current post on your dashboard (to keep your place during tab crashes and refreshes). **//
+//* FRAME false **//
+//* BETA false **//
+//* SLOW false **//
+
+XKit.extensions.addressbar = new Object({
+
+	running: false,
+	interval: 0, //id from setInterval
+	initialScroll: null, //initial value from history.scrollRestoration
+	
+	preferences: {
+		"focus_only": {
+			text: "Only run the extension when the page has focus",
+			default: true,
+			value: true
+		},
+	},
+
+	run: function() {
+		this.running = true; //there's some confusion over whether this gets set to true on immediate return
+		
+		const Xwhere = XKit.interface.where();
+		if (Xwhere.dashboard !== true || Xwhere.endless !== true) {
+			return; //only useful on endless dashboard
+		} 
+
+		this.interval = setInterval(this.update_address, 500);
+
+		this.initialScroll = history.scrollRestoration;
```

Comment:
this is only intended to be called once, correct? it would make more sense to store it on the extension object directly, rather than initialise the key as `null` and then set it upon every activation - which considering how it's used in the `destroy()` logic, effectively does nothing on subsequent calls